### DISCUSSION TITLE
files/dirs beginning with `$` does not work in Neovim.

### DISCUSSION THREAD

**Discussion by molleweide** [remix-run/remix#1778] [Q&A] [answered]
Hi there,

Files starting with `$` does not seem to work for me in neovim. I have tried `\$myId.tdx` and `$myId.tsx` and both of these are renamed automatically `.tsx`. 

Does anyone here have a workaround for this? Is there maybe another string prefix one can use instead of `$` for dynamic routes.

According to one discord use putting a backslash in the beginning of the file name "should" work but it does not work for me.

Also, if somebody doesn't mind, could you point me to where it is documented how none creates programmatic routes instead.
Because I would very much like to learn `remix` but if I cannot use `$` in beginning of my file names then I am not
sure how to get around this atm.

**Comment by molleweide** [comment]
A follow up question. This convention of naming files with `$`, is this a remix convention or more of a react thingy?

**Reply by sergiodxa** [reply]
It's a Remix convention, or frameworks uses things like `[param]` but Remix decided to use `[]` to escape characters in the URL, so you can do `routes/rss[.].xml.ts` and get the URL `/rss.xml`, the `[.]` escapes the `.` so it's part of the URL and not a path separator, without that doing `routes/rss.xml.ts` would give you the URL `/rss/xml`.

**Comment by sergiodxa** [comment, marked as answer]
If you editor doesn't support `$` in the file names and changing it's not an option you can define your routes outside the routes folder and use the routes function in the remix.config.js file to define your routes with code https://remix.run/docs/en/v1/api/conventions#routes
